[Cedric Lim Jun Wei] iP

Cedo8 opened the pull request.

Commit:
Add Gradle support

Cedo8 committed:
Added greeting, echoing and exit functionality to Duke.java

Cedo8 committed:
Added add and list functionality to Duke.java

Cedo8 committed:
Added Mark as Done functionality to Duke.java
Added Task.java

Cedo8 committed:
Added ability to categorise Tasks (ToDos, Events and Deadlines) to Duke.java
Updated toString for Task.java
Added ToDo.java, Event.java and Deadline.java

Cedo8 committed:
Added test cases (input.txt) and their expected results (EXPECTED.txt)
Carried out UI testing using runtest.bat

Cedo8 committed:
Added error handling functionality to Duke.java
Added DukeException.java
Updated test cases (input.txt) and expected results (EXPECTED.txt)

Cedo8 committed:
Added delete functionality to Duke.java
Updated test cases (input.txt) and expected results (EXPECTED.txt)

Cedo8 committed:
Add duke.text to store saved daata
Add save functionality to Duke.java
Add new constructors to ToDo.java, Deadline.java and Event.java

Cedo8 committed:
Add date and time functionality for deadlines in Duke.java

Cedo8 committed:
Merge branch 'branch-Level-7'

Cedo8 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Duke.java

Cedo8 committed:
Extract UI, storage, parsing and tasklist functionality from Duke.java
Add Ui.java, Storage.java, Parser.java, TaskList.java

Cedo8 committed:
Create package duke to store all classes
Create subpackage duke.task to store task classes

Cedo8 committed:
Add Junit Tests

Cedo8 committed:
Publish iP v0.1 as jar file
Update path directory in Duke.java
Update file creation in Storage.java

Cedo8 committed:
Update JavaDoc comments for all classes

Cedo8 committed:
Update code to follow required coding standards

Cedo8 committed:
Add find functionality to Parser.java

Cedo8 committed:
Merge branch 'branch-A-JavaDoc'

Cedo8 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/task/ToDo.java

Cedo8 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java

koonweee commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,95 @@
+     * @return ArrayList of all existing tasks
+     * @throws IOException if filePath does not exist
+     */
+    public ArrayList<Task> load() throws IOException {
```
Could separate logical units within a block by a blank line

koonweee reviewed (commented):
Overall looks to adhere to Java coding standard. Only minor issue would be perhaps to add more line breaks to improve readability.

Cedo8 committed:
Merge branch 'add-gradle-support' of https://github.com/Cedo8/ip into branch-A-Gradle

Cedo8 committed:
Add Gradle support

Cedo8 committed:
Merge branch 'branch-A-Gradle'

# Conflicts:
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

Cedo8 committed:
Update Duke.java, Parser.java and Storage.java to satisfy checkstyle requirements
Add checkstyle.xml and suppressions.xml

Cedo8 committed:
Update build.gradle, Duke.java, Parser.java and Storage.java to work with JavaFX
Add DialogBox.java, Launcher.java, Main.java, MainWindow.java, DialogBox.fxml and MainWindow.fxml to work with JavaFX

Cedo8 committed:
Merge branch 'branch-Level-10'

CodingCookieRookie commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,274 @@
+     * @return duke's reply
+     * @throws IOException if filePath does not exist
+     */
+    public static String parse(String command, TaskList tasks, Storage storage, String output) throws IOException {
```
Can perhaps abstract the different commands to avoid a long command? :)

CodingCookieRookie commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,118 @@
+     *
+     * @throws IOException if filePath does not exist
+     */
+    public void clear() throws IOException {
```
I really like how you included the clear() method and the code is very clean

CodingCookieRookie commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,42 @@
+     * @return task in string form
+     */
+    @Override
+    public String toString() {
```
Perhaps since this is a multi-part customised line you could use String.format instead?

CodingCookieRookie commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,75 @@
+    }
+
+    /**
+     * Alternative constructor for empty list
```
Perhaps should put a fullstop after the sentence? Found this issue at several places as well

CodingCookieRookie reviewed (commented):
Generally very well done, with accurate styles and indentation.

Cedo8 committed:
Fix bug where long messages cannot be fully displayed
Add new jar release with GUI implementation

Cedo8 committed:
Abstract methods in Parser.java to adhere to required code quality
Update Duke.java and Storage.java to adhere to required code quality

Cedo8 committed:
Add assertions to Parser.java, Storage.java and TaskList.java

Cedo8 committed:
Merge pull request #1 from Cedo8/branch-A-Assertions

Add assertions to Parser.java, Storage.java and TaskList.java

Cedo8 committed:
Merge branch 'master' of https://github.com/Cedo8/ip

Cedo8 committed:
Merge branch 'master' into branch-A-CodeQuality

Cedo8 committed:
Merge pull request #2 from Cedo8/branch-A-CodeQuality

Abstract methods in Parser.java to adhere to required code quality

Cedo8 committed:
Merge branch 'master' of https://github.com/Cedo8/ip

Cedo8 committed:
Add duplicate detection functionality to Parser.java

Cedo8 committed:
Merge branch 'branch-A-C-DetectDuplicates'

Cedo8 committed:
Update GUI visuals

Cedo8 committed:
Set theme jekyll-theme-merlot

Cedo8 committed:
Add user guide and Ui screenshots

Cedo8 committed:
Update filepath in Main.java
Update user guide
Update task marking system
Add new response to no matching results for 'find'

Cedo8 committed:
Update user guide

Cedo8 committed:
Update user guide

Cedo8 committed:
Add functionality to exit application with command
Update README.md and Ui.png
Update GUI

Cedo8 committed:
Rename Ui.PNG to Ui.png

Cedo8 committed:
Rename Ui.PNG to Ui.png

Cedo8 committed:
Rename Ui.PNG to Ui.png
